WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSectionHeader.cs

```diff
@@ -87,39 +87,193 @@ public class WiseSectionHeader : BaseBinaryDeserializer<SectionHeader>
+            // TODO: Count size of string section for later size verification
+            
+            data.Seek(initialOffset + localWisOffset, 0);
+            header.TmpString = data.ReadNullTerminatedString(Encoding.ASCII); // read .TMP string
+            header.GuidString = data.ReadNullTerminatedString(Encoding.ASCII); // read GUID string
+            // TODO: better way to figure out how far it's needed to advance?
+            if (header.Version == null)
+            {
+                // TODO: error output
+                return null;
+            }
+            int versionSize;
+            if (header.Version[header.Version.Length - 1] == 0x02)
+            {
+                versionSize = header.Version[header.Version.Length - 3];
+            }
+            else
+            {
+                versionSize = header.Version[header.Version.Length - 2];
+            }
+            if (versionSize != 1) // third byte seems to indicate size of NonWiseVer
+            {
 
-            header.UnknownValue9 = data.ReadUInt32LittleEndian();
-            header.UnknownValue10 = data.ReadUInt32LittleEndian();
-            header.UnknownValue11 = data.ReadUInt32LittleEndian();
-            header.UnknownValue12 = data.ReadUInt32LittleEndian();
-            header.UnknownValue13 = data.ReadUInt32LittleEndian();
-            header.UnknownValue14 = data.ReadUInt32LittleEndian();
-            header.UnknownValue15 = data.ReadUInt32LittleEndian();
-            header.UnknownValue16 = data.ReadUInt32LittleEndian();
-            header.UnknownValue17 = data.ReadUInt32LittleEndian();
-            if (headerLength == 17)
-                return header;
+                byte[] stringBytes = data.ReadBytes(versionSize);
+                header.NonWiseVersion = Encoding.ASCII.GetString(stringBytes);
+                if (localWisOffset <= 77)
+                {
+                    header.PreFontValue = data.ReadBytes(2);
+                }
+                else
+                {
+                    header.PreFontValue = data.ReadBytes(4);
+                }
+            }
+            else // If that third byte is 0x01, no NonWiseVersion string is present.
+            {
+                header.PreFontValue = data.ReadBytes(3);
+            }
+            header.FontSize = data.ReadByte(); 
+            int preStringBytesSize = WiseSectionPreStringBytesSize[localWisOffset];
+            if (header.Version[1] == 0x01)
+            {
+                preStringBytesSize = 2; // hack for Codesited5.exe , very early and very strange.
+            }
+            header.StringValues = data.ReadBytes(preStringBytesSize);
+            List<byte> stringList = new List<byte>(); // List of string bytes to be set to final value
+            int counter = 0;
+            bool endNow = false;
+            bool languageSection = false;
+            int languageSectionCounter = 0;
+            while (counter < preStringBytesSize) // Iterate pre-string byte array
```

Comment:
Is there any case where this loop could iterate without the counter incrementing? I tried reading through a few times but the branches made it difficult to follow.

Reply:
As you've probably noticed, my parsing code here is awful, and trying to deal with this is what made me give up on actually finishing all the stuff I wanted to for this PR. After I implement some of your other fixes, I'll try and give you a basic outline of what it is I'm actually trying to do here.